src/EventBus.Dapr/DaprEventBusEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Dapr.Client;
+using EventBus.Abstractions;
+using EventBus.Dapr;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="IEndpointRouteBuilder" />.
+    /// </summary>
+    public static class DaprEventBusEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Maps endpoints that will handle requests for DaprEventBus subscriptions.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder" />.</param>
+        /// <param name="configure">The original <see cref="T:Microsoft.AspNetCore.Builder.IApplicationBuilder" />.</param>
+        /// <returns>An <see cref="DaprEventBusEndpointConventionBuilder"/> for endpoints associated with DaprEventBus subscriptions.</returns>
+        public static DaprEventBusEndpointConventionBuilder MapDaprEventBus(this IEndpointRouteBuilder endpoints,
+            Action<IEventBus> configure = null)
+        {
+            if (endpoints is null)
+                throw new ArgumentNullException(nameof(endpoints));
+
+            var logger = endpoints.ServiceProvider.GetService<ILogger<DaprEventBus>>();
+            var eventBus = endpoints.ServiceProvider.GetService<IEventBus>();
+            var daprClient = endpoints.ServiceProvider.GetService<DaprClient>();
+            var eventBusOptions = endpoints.ServiceProvider.GetService<IOptions<DaprEventBusOptions>>();
+
+            // Configure event bus
+            logger.LogInformation($"Configuring event bus ...");
+            configure?.Invoke(eventBus);
+
+            endpoints.MapSubscribeHandler();
+            IEndpointConventionBuilder builder = null;
+            foreach (var topic in eventBus.Topics)
+            {
+                logger.LogInformation($"Mapping Post for topic: {topic.Key}");
+                builder = endpoints.MapPost(topic.Key, HandleMessage)
```

Comment:
Since we are mapping multiple endpoints, only the builder for the last endpoint added will return from `MapDaprEventBus`.